packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -885,6 +890,23 @@ export class GarbageCollector implements IGarbageCollector {
             this.runtime.deleteUnusedRoutes(gcResult.deletedNodeIds);
         }
 
+        // If we are running in GC tombstone mode, tombstone objects for unused routes. This enables testing scenarios
+        // involving access to "deleted" data without actually deleting the data from summaries.
+        // Note: we will not tombstone in test mode
+        if (this.tombstoneMode && !this.testMode) {
```

Comment:
nit: make this an else of this.testMode and you can remove the check for !this.testMode.